data/libs/SpaceStation.lua

```diff
@@ -711,15 +838,26 @@ Event.Register("onGameStart", function ()
 				end
 			end
 		end
+
+		-- SAVEBUMP: fixup for save version 87 where player is docked to station without visited data
+		local dockedStation = Game.player:GetDockedWith()
+		if dockedStation and not visited[dockedStation] then
+			createStationData(dockedStation)
+		end
 		loaded_data = nil
 	end
 
 	Timer:CallEvery(3600, updateSystem)
 end)
 
 Event.Register("onShipDocked", function (ship, station)
-	if ship ~= Game.player or SpaceStation.adverts[station] then return end
-	createStationMarket(station)
+	if ship ~= Game.player then return end
+
+	if not visited[station] then
+		createStationData(station)
+	else
+		updateStationMarket(station)
```

Comment:
hmm, this is updated only when re-docking, it turns out if the player wants to stay longer at the station, he will see that nothing is happening to the market?
and if it then undocks and docks, there will be a sharp change in the market?

Reply:
There is some possibility to observe a change in the market when redocking, though given that updates happen on a timescale of in-game weeks it takes a non-trivial amount of time even sitting on a pad at max time acceleration to reach even a single update tick, and said single tick has a comparatively small effect on the amount of stock available at the station.

Reply:
Part of my hope is to discourage this behavior and encourage jumping out of the system or trading multiple commodities instead - the average player will not know the precise details of when the market restocks and at what interval, only that it happens sometime during their travel away from the station.

Additionally, I got entirely sidetracked by the discussion because updateSystem already does this and updates all stations each hour.